### ISSUE TITLE
[Bug]: TypeFieldEnum not found

### ISSUE THREAD

**Issue by billmn** [joaopaulolndev/filament-edit-profile#6]
### What happened?

If custom fields is enabled, the edit profile page show this exception related to **CustomFieldsForm** class:

```
Class "Joaopaulolndev\FilamentGeneralSettings\Enums\TypeFieldEnum" not found
```

### How to reproduce the bug

Enable and add custom fields in **filament-edit-profile.php config** file, after that open the edit profile page.

### Package Version

1.0.5

### PHP Version

8.2.18

### Laravel Version

11.8.0

**Comment by joaopaulolndev** [comment]
Hello @billmn,

I just conducted a test and couldn't reproduce the error. Here is the configuration file from my example:


`
<?php

return [
    'show_custom_fields' => true,
    'custom_fields' => [
        'address' => [
            'type' => 'text',
            'label' => 'Endereço',
            'placeholder' => 'Informe o endereço',
            'required' => true,
            'rules' => 'required|string|max:255',
        ],
    ]
];
`

Please note that I no longer use an enum for the 'type'. Instead, I'm specifying the type directly as a string:

`
'type' => 'text',
`

![image](https://github.com/joaopaulolndev/filament-edit-profile/assets/11706937/09b1008c-7607-4b6c-ba17-175371709577)

Could you provide more details about where you are encountering this error? It might help to remove and reinstall the package, as there could be a caching issue in your vendor directory.

Please let me know if the issue persists after reinstalling the package.

I hope this helps! Let me know if you need any further adjustments.

**Comment by billmn** [comment]
Hi @joaopaulolndev,
I have already tried to reinstall the package but the issue persist.

I notice that the CustomFieldsForm class still uses the Enum in tag **1.0.5** :
https://github.com/joaopaulolndev/filament-edit-profile/blob/b04f58eac0b9b8f0770da12ac3781a4e8c31985a/src/Forms/CustomFieldsForm.php

**Comment by joaopaulolndev** [comment]
Hey @billmn , 

I already fixed it in v1.0.6, please update the package and try again.
Please let me know if now it's working fine.

I will waiting for your answer before closing this issuer :)

Many thanks for reporting it!

**Comment by billmn** [comment]
Now it works thank you.

However I get an exception when I click "Save" in the custom fields form as it tries to save the field in the user table instead of just saving it in the "custom_fields" field.

This is the exception message (extension is the name of my field)
```SQLSTATE[42S22]: Column not found: 1054 Unknown column 'extension' in 'field list'```

Credo che il problema sia qui:
<img width="586" alt="Screenshot 2024-05-25 alle 16 06 32" src="https://github.com/joaopaulolndev/filament-edit-profile/assets/779534/0fd09625-4a0f-4b66-adce-6a1c194caf20">

**Comment by joaopaulolndev** [comment]
Did you publish and ran the migration ?

```
php artisan vendor:publish --tag="filament-edit-profile-migrations"
php artisan migrate
```

Also put the field custom_fields in the $fillable variable in your User model?

```
protected $fillable = [
    'name',
    'email',
    'password',
    'custom_fields',
];
```

The last one, put the variable in casts?
```
protected function casts(): array
{
    return [
        'email_verified_at' => 'datetime',
        'password' => 'hashed',
        'custom_fields' => 'array'
    ];
}
```

**Comment by billmn** [comment]
Yes, but the problem is a different one:
<img width="708" alt="Screenshot 2024-05-25 alle 16 17 41" src="https://github.com/joaopaulolndev/filament-edit-profile/assets/779534/06daee9b-6d9a-4301-addc-18635cc6c2f1">

As you can see from the query, it is trying to save the value in the "extension" field

**Comment by joaopaulolndev** [comment]
@billmn please could you send me a print from your form displaying?
It could be one thing but I'm not right yet because I'm not able to reproduce.


Might you put a dd() here, you can see the values inside the array. 
may need to unset the custom variables. But, let's see your print

![image](https://github.com/joaopaulolndev/filament-edit-profile/assets/11706937/9e70b452-bc79-4890-8a1f-494a18b6e48b)

**Comment by billmn** [comment]
Sure

<img width="687" alt="Screenshot 2024-05-25 alle 16 35 13" src="https://github.com/joaopaulolndev/filament-edit-profile/assets/779534/871a59db-6db9-4bdb-ab52-ed42f0ecf3a2">

This is the config file:
```
return [
    'show_custom_fields' => true,
    'custom_fields' => [
        'extension' => [
            'type' => 'text',
            'label' => 'Extension',
            'rules' => 'string|max:255',
            'required' => false,
            'placeholder' => '',
        ],
    ],
];
```

**Comment by joaopaulolndev** [comment]
Please change inside the vendor the method where you put @dd to:

```
    public function updateCustomFields(): void
    {
        try {
            $data = $this->customFieldsForm->getState();
            $data['custom_fields'] = $data ?? [];

            $customFields['custom_fields'] = $data['custom_fields'];
            $this->handleRecordUpdate($this->getUser(), $customFields);
        } catch (Halt $exception) {
            return;
        }

        $this->sendSuccessNotification();
    }
```

Let me know with you fix the issue if fixed I put it in a new release

**Comment by billmn** [comment]
Yes, with this change it works, thank you.

**Comment by joaopaulolndev** [comment]
Please run the composer update, which was fixed in v1.0.7
Thank you so much for your help
